extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/activeconnection/ActiveConnectionDirectory.java

```diff
@@ -44,34 +44,40 @@ public class ActiveConnectionDirectory extends RestrictedObject
 
     @Override
     public ActiveConnection get(String identifier) throws GuacamoleException {
+        validateUserAccess();
```

Comment:
I don't really love having to call this everywhere, but I'm not sure of a better / more elegant way to do this.

Reply:
What about a `RestrictedDirectory` base class?

Reply:
That sounds like a good idea - I'll give it a try!

Reply:
Ok, how does this look?

Reply:
Looks much more DRY, I'd say.